From 62a1fe74800bfe65b81e3aa0e307d1856c7ffc03 Mon Sep 17 00:00:00 2001
From: "sheng.liu" <sheng.liu@amlogic.com>
Date: Wed, 22 Dec 2021 12:32:12 +0800
Subject: [PATCH] gstreamer1.0-plugins-good: compare pts with valid value 
 [1/1]

PD#SWPL-64204

Problem:
pts compare between invalid pts and valid pts which will got the wrong frame order

Solution:
check pts if it's valid before compare

Verify:
AP222
---
 sys/v4l2/gstv4l2videodec.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/sys/v4l2/gstv4l2videodec.c b/sys/v4l2/gstv4l2videodec.c
index 735881d..24dc1ba 100755
--- a/sys/v4l2/gstv4l2videodec.c
+++ b/sys/v4l2/gstv4l2videodec.c
@@ -451,7 +451,10 @@ gst_v4l2_video_dec_get_oldest_frame (GstVideoDecoder * decoder)
   for (l = frames; l != NULL; l = l->next) {
     GstVideoCodecFrame *f = l->data;
 
-    if (!frame || frame->pts > f->pts)
+    if (!frame
+    || (GST_CLOCK_TIME_IS_VALID(frame->pts)
+       && GST_CLOCK_TIME_IS_VALID(f->pts)
+       && (frame->pts > f->pts)))
       frame = f;
 
     count++;
-- 
2.32.0

